src/content/app/tools/blast/components/listed-blast-submission/ListedBlastSubmission.tsx

```diff
@@ -143,30 +146,52 @@ const SequenceBox = (props: SequenceBoxProps) => {
       <div>Sequence {sequence.id}</div>
       <div className={styles.sequenceHeader}>{sequenceHeaderLabel}</div>
       <div className={styles.speciesCount}>
-        <span className={styles.againstText}>Against</span>
-        {speciesCount} species
+        <span className={styles.againstText}>Against</span> {speciesCount}{' '}
+        species
       </div>
-      <StatusElement jobs={jobs} />
+      {jobs.length ? (
+        <StatusElement sequenceId={sequence.id} jobs={jobs} />
+      ) : null}
     </div>
   );
 };
 
-const StatusElement = ({ jobs }: { jobs: BlastJob[] }) => {
-  const hasRunningJobs = jobs.some((job) => job.status === 'RUNNING');
-  const hasFailedJobs = jobs.some((job) => job.status === 'FAILURE');
+type StatusElementProps = {
+  sequenceId?: number;
+  jobs: BlastJob[];
+};
+
+export const StatusElement = (props: StatusElementProps) => {
+  const { sequenceId, jobs } = props;
+
+  isFailedBlastSubmission();
+
+  const hasRunningJobs = jobs.some(
+    (job) =>
+      job.status === 'RUNNING' &&
+      (sequenceId ? sequenceId === job.sequenceId : true)
```

Comment:
Why are we doing this sequence id check in this component? Why are we not passing it only relevant jobs from the parent?

Reply:
Removed them as we dont need to do this check.